unittests/backends/quantinuum/QuantinuumStartServerAndTest.sh.in

```diff
@@ -47,17 +47,64 @@ fi
 rm -f @CMAKE_BINARY_DIR@/FakeCppQuantinuum.config
 echo "key: key" >> @CMAKE_BINARY_DIR@/FakeCppQuantinuum.config
 echo "refresh: refresh" >> @CMAKE_BINARY_DIR@/FakeCppQuantinuum.config
-# Run the test for the QIR output
-@CMAKE_BINARY_DIR@/bin/nvq++ -fenable-cudaq-run --target quantinuum --quantinuum-credentials @CMAKE_BINARY_DIR@/FakeCppQuantinuum.config --quantinuum-machine Helios-1SC --quantinuum-project mock --quantinuum-url http://localhost:62440 @CMAKE_SOURCE_DIR@/unittests/backends/quantinuum/test_run.cpp -o test_run.x
-CUDAQ_ENABLE_QUANTUM_DEVICE_RUN=1 ./test_run.x
 
-# Did they fail?
-testsPassed=$?
+# Run all the nvq++ tests with mock server.
+echo "### nvq++ tests"
+set +e # Allow script to keep going through errors
+test_err_sum=0
+for filename in @CMAKE_SOURCE_DIR@/unittests/backends/quantinuum/test_*.cpp; do
+  echo "Testing $filename"
+  @CMAKE_BINARY_DIR@/bin/nvq++ --target quantinuum --quantinuum-credentials @CMAKE_BINARY_DIR@/FakeCppQuantinuum.config --quantinuum-machine Helios-1SC --quantinuum-project mock --quantinuum-url http://localhost:62440 $filename 
+  test_status=$?
+  if [ $test_status -eq 0 ]; then
+    ./a.out
+    test_status=$?
+    if [ $test_status -eq 0 ]; then
+      echo "Successfully ran test: $filename"
+    else
+      echo "Test failed (failed to execute): $filename" 
+      test_err_sum=$((test_err_sum+1))
+    fi
+  else
+    echo "Test failed (failed to compile): $filename" 
+    test_err_sum=$((test_err_sum+1))
+  fi
+done
+
+# Test separate compilation
+# Compile device code lib (with -fPIC -shared)
+@CMAKE_BINARY_DIR@/bin/nvq++ -fPIC -shared --target quantinuum --quantinuum-credentials @CMAKE_BINARY_DIR@/FakeCppQuantinuum.config --quantinuum-machine Helios-1SC --quantinuum-project mock --quantinuum-url http://localhost:62440 @CMAKE_SOURCE_DIR@/unittests/backends/quantinuum/run_separate_compilation/lib.cpp -o lib.so
```

Comment:
Why is the library compiled with a certain target? I think it shouldn't have to be compiled against a specific target and we may even want to give an error if a target is set when `-shared` is passed (and maybe a warning when `-shared` is passed but not -fPIC?).